### PR TITLE
feat: routing with `react-router@6` and sugar

### LINKED ISSUE #209: Upgrade to react-router@6
Several changes need to be made to upgrade to the latest React Router.

## Rewrite `<Link>` usage

Use function to dynamically build the to path.

### Problem

react-router@6 does not allow nested <Router> elements.  We are using a loop hole in v5 to specify a conditional basename on a child <Router> to force child <Link>s to dynamically include a sidechain’s url.

Ex. `<Link to="/network"/>` resolves to `<a href="/example.mysidechain.com/network">` because it is inside the nested `<Router basename="example.mysidechain.com">`.

### Design

Since we are already rewriting every place using<Link>we should implement a typed route system. 

```
const function build<T>(route: Route<T>, params: T) { 
  const basename = sidechainUrl ?? '';
  generatePath(
    `${basename}/${route.path}`,
    params
  )
}
```

### Inspiration

https://blog.codacy.com/migrating-to-react-typed-named-routes-react-router-typescript/ 

### Problem

In react-router@6 match and location are no longer auto populated when used in conjunction with a <Route>.  With the removal of withRouter there isn’t another way to populate them besides the dreaded prop drilling.

### Design

Leverage hooks like `useParams`, `useLocation`, and `useMatch`. This can be done before switching to v6.

### Components Affected

- `LedgersPage`: Ledegers/index.js
- `Ledger`: Ledger/index.js
- `Transaction`: Transactions/index.js
- `Network`: Network/index.js
- `Validator`: Validators/index.js
- `PayString`: PayString/index.js
- `Token`: Token/index.js
- `NoMatch`: NoMatch/index.js

### Components Affected

- `App`: App/App.js
- `Search`: Header/Search.jsx

## Combine `AppWrapper` and `App`

There is no need to separate them now that nested Routers aren't required and most of its logic revolves around that.

### Design

Update App/index.js to be just an import point. Move logic into App/App.jsx.  Bonus points if we switch it to Typescript.

### LINKED ISSUE #602: Persist websocket connections across page transitions
Websocket connections are reset when pressing the browser back button.  Something to be considered for #601

### PR DESCRIPTION
**Pull request by ckniffen**
## High Level Overview of Change

- Add sugar perform type checking and to build paths when running "custom" mode.
- Upgrade to `react-router@6`
- Fix active navigation menu highlighting.  Was only counting some pages as being in the explorer.
- Fix #602 allowing for faster back navigation

Closes #209 and fixes #602

### Context of Change

This eliminates the nested router workaround which should not have worked to begin and made testing much more complicated.  This is done by building the path dynamically based on custom mode.

Many of the tests required updating because the position of `title` and `class` attributes on anchors created by react-router's `<Link>` changed order.

### Type of Change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Refactor (non-breaking change that only restructures code)
- [x] Tests (You added tests for code that already exists, or your new feature included in this PR)
- [x] Documentation Updates

### CODE REVIEW

**Review by pdp2121 — APPROVED** [review]

**Review by justinr1234 — APPROVED** [review]

File: src/containers/Transactions/test/Meta.test.tsx [hunk 1]
```diff
@@ -32,35 +32,35 @@ describe('TransactionMeta container', () => {
     expect(w.find('li').length).toBe(23)
 
     expect(w.find('li').at(0).html()).toBe(
-      '<li class="meta-line">owned_account_root <a class="account" title="rUmustd4TbkjaEuS7S1damozpBEREgRz9z" href="/accounts/rUmustd4TbkjaEuS7S1damozpBEREgRz9z">rUmustd4TbkjaEuS7S1damozpBEREgRz9z</a><ul class="meta-line"><li>Balance decreased by<b>-\uE90050.324316<small>XRP</small></b>from<b>\uE9002,910.704988<small>XRP</small></b>to<b>\uE9002,860.380672<small>XRP</small></b></li></ul></li>',
+      '<li class="meta-line">owned_account_root <a title="rUmustd4TbkjaEuS7S1damozpBEREgRz9z" class="account" href="/accounts/rUmustd4TbkjaEuS7S1damozpBEREgRz9z">rUmustd4TbkjaEuS7S1damozpBEREgRz9z</a><ul class="meta-line"><li>Balance decreased by<b>-\uE90050.324316<small>XRP</small></b>from<b>\uE9002,910.704988<small>XRP</small></b>to<b>\uE9002,860.380672<small>XRP</small></b></li></ul></li>',
```
Comment (jonathanlei) [inline src/containers/Transactions/test/Meta.test.tsx:35]: why did the title and class change order?

Comment (ckniffen) [reply src/containers/Transactions/test/Meta.test.tsx:35]: `react-router` changed them for `<Link>`'s output.

Comment (ckniffen) [reply src/containers/Transactions/test/Meta.test.tsx:35]: It is referenced in "Context of Change"

**Review by jonathanlei — APPROVED** [review]
LGTM

**Review by pdp2121 — APPROVED** [review]

**Review by justinr1234 — APPROVED** [review]
Only nits, looks good!

**Review by justinr1234 — APPROVED** [review]

**Pull request merged**